internal/guest/storage/devicemapper/devicemapper.go

```diff
@@ -134,9 +120,8 @@ func (err *dmError) Error() string {
 
 // ioctl issues the specified device-mapper ioctl
 func ioctl(f *os.File, code int, data *dmIoctl) error {
-	_, _, errno := unix.Syscall(unix.SYS_IOCTL, f.Fd(), uintptr(code|_DM_IOCTL_BASE), uintptr(unsafe.Pointer(data)))
-	if errno != 0 {
-		return &dmError{Op: code, Err: errno}
+	if err := linux.Ioctl(f, code|_DM_IOCTL_BASE, uintptr(unsafe.Pointer(data))); err != nil {
```

Comment:
We need to be careful with how we pass arguments here. It's only safe to pass a Go-pointer-converted-to-`uintptr` if it's passed directly to `unix.Syscall`[1]. So by converting to `uintptr` here and then calling a function which in turn calls `unix.Syscall`, we are at risk if memory is rearranged[2].

[1] See rule 4 here: https://pkg.go.dev/unsafe#Pointer. Technically the requirement is actually that the conversion must happen directly in the call to any function implemented in assembly (as `unix.Syscall` is), or that is marked with `//go:uintptrescapes` (see [here](https://pkg.go.dev/cmd/compile).

[2] While Go's GC does not do any memory rearranging today, there is no explicit promise that it won't start to do so in the future. Additionally, there is still risk if the passed pointer is to a stack value, and the stack is resized after the address is taken.